packages/teams-js/src/public/app.ts

```diff
@@ -63,88 +67,93 @@ export function appInitializeHelper(apiVersionTag: string, validMessageOrigins?:
-          } catch (e) {
-            if (e instanceof SyntaxError) {
-              try {
-                initializeHelperLogger('Attempting to parse %s as an SDK version', runtimeConfig);
-                // if the given runtime config was actually meant to be a SDK version, store it as such.
-                // TODO: This is a temporary workaround to allow Teams to store clientSupportedSDKVersion even when
-                // it doesn't provide the runtimeConfig. After Teams updates its client code, we should
-                // remove this feature.
-                if (!isNaN(compareSDKVersions(runtimeConfig, defaultSDKVersionForCompatCheck))) {
-                  GlobalVars.clientSupportedSDKVersion = runtimeConfig;
-                }
-                const givenRuntimeConfig: IBaseRuntime | null = JSON.parse(clientSupportedSDKVersion);
-                initializeHelperLogger('givenRuntimeConfig parsed to %o', givenRuntimeConfig ?? 'null');
-
-                if (!givenRuntimeConfig) {
-                  throw new Error(
-                    'givenRuntimeConfig string was successfully parsed. However, it parsed to value of null',
-                  );
-                } else {
-                  applyRuntimeConfig(givenRuntimeConfig);
-                }
-              } catch (e) {
-                if (e instanceof SyntaxError) {
-                  applyRuntimeConfig(
-                    generateVersionBasedTeamsRuntimeConfig(
-                      GlobalVars.clientSupportedSDKVersion,
-                      versionAndPlatformAgnosticTeamsRuntimeConfig,
-                      mapTeamsVersionToSupportedCapabilities,
-                    ),
-                  );
-                } else {
-                  throw e;
-                }
+      GlobalVars.initializePromise = initializeCommunication(
+        validMessageOrigins,
+        apiVersionTag,
+        customMessageIdStarter,
+      ).then(({ context, clientType, runtimeConfig, clientSupportedSDKVersion = defaultSDKVersionForCompatCheck }) => {
+        GlobalVars.frameContext = context;
+        GlobalVars.hostClientType = clientType;
+        GlobalVars.clientSupportedSDKVersion = clientSupportedSDKVersion;
+        // Temporary workaround while the Host is updated with the new argument order.
+        // For now, we might receive any of these possibilities:
+        // - `runtimeConfig` in `runtimeConfig` and `clientSupportedSDKVersion` in `clientSupportedSDKVersion`.
+        // - `runtimeConfig` in `clientSupportedSDKVersion` and `clientSupportedSDKVersion` in `runtimeConfig`.
+        // - `clientSupportedSDKVersion` in `runtimeConfig` and no `clientSupportedSDKVersion`.
+        // This code supports any of these possibilities
+
+        // Teams AppHost won't provide this runtime config
+        // so we assume that if we don't have it, we must be running in Teams.
+        // After Teams updates its client code, we can remove this default code.
+        try {
```

Comment:
I am not sure why all of this showed up as changed in this PR. Only `customMessageIdStarter` is added to some placed and that was it